vision/app.py

```diff
@@ -133,6 +156,7 @@ def do_image(self, im):
             goal = None
 
             if self.lock_id is None:
```

Comment:
Maybe not here that this only occurs when the goal is first locked upon (that's when `lock_id` == None). Otherwise, we try to find the goal that matches the `lock_id`. The purpose of all this `lock` crap is so that if the target we're driving to is no longer the center-most blob, we don't start driving to a different target since we're constantly updating the path (in 2019 game, anyways).